src/cobra/test/test_core/test_core_reaction.py

```diff
@@ -18,6 +18,16 @@ def test_gpr() -> None:
     model = Model()
     reaction = Reaction("test")
 
+    # Set GPR to an empty string
+    reaction.gene_reaction_rule = ""
+    # Empty gene_reaction_rule leads to an empty GPR
+    assert hasattr(reaction._gpr, "body")
+    assert reaction.gpr.to_string() == ""
+    assert reaction.gpr.to_string(names={"goo": "blah"}) == ""
+    # Set GPR directly (shouldn't really be used, but just a test)
+    reaction.gpr = GPR()
+    assert reaction.gene_reaction_rule == ""
+    assert reaction.gpr.eval()
```

Comment:
As I mentioned above, I think only having the `Reaction.gene_reaction_rule` interface is enough and there is no need at this point to also introduce `Reaction.gpr`.